modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/di/ModelModule.kt

```diff
@@ -39,3 +65,11 @@ object ModelModule {
     @Provides
     fun provideExternalDataDao(database: AppDatabase): ExternalDataDao = database.externalDataDao()
 }
+
+@Qualifier
+@Retention(AnnotationRetention.BINARY)
+annotation class RoomConverters
+
+fun <T : RoomDatabase> RoomDatabase.Builder<T>.addTypeConverters(converters: List<Any>) = converters.fold(this) { builder, converter ->
```

Comment:
It took me a while to get my head around the `fold` command. Is this a common pattern if you want to return `this` as the result?

Reply:
It's actually not returning `this`. It's returning whatever is the result of the lambda in `fold()`. So

```kotlin
val list = listOf(1, 2, 3, 4, 5)
val result = list.fold("") { str, int -> "$str$int" }
println(result) // "12345" 
```

It doesn't matter much in this instance because `Builder` is most likely mutable. I haven't checked but I assume so. So these two are equivalent.

```kotlin
converters.forEach { converter -> 
  builder.add(converter)
}
return builder

return converters.fold(builder) { builder, converter -> builder.add(converter) } 
```

But it does matter when object which we fold upon is immutable.

Whether it is common it depends on who you ask I guess. It's built into the standard library. It's super common and natural for me. In functional programming it is a standard way.

Reply:
Thanks for taking the time to explain that to me.